shared/src/commonMain/kotlin/io/customer/shared/tracking/api/model/BatchTracking.kt

```diff
@@ -0,0 +1,36 @@
+package io.customer.shared.tracking.api.model
+
+import io.customer.shared.tracking.model.TrackingError
+import kotlinx.serialization.SerialName
+
+/**
+ * Request class only to be used when making tracking requests in batch.
+ */
+@kotlinx.serialization.Serializable
+internal data class BatchTrackingRequestBody(
+    @SerialName("batch") val batch: List<TrackingRequest>,
```

Comment:
```suggestion
    val batch: List<TrackingRequest>,
```

I don't see a need for `@SerialName()` annotation

Reply:
I don't see any harm in this. I have explained my [idea behind this before](https://github.com/customerio/mobile-shared/pull/1#discussion_r1022914960).

Let's keep it as it is for this PR, as we have this consistency throughout the project. I'll add this to open discussions and if we decide to do this change, we'll do in a single separate PR.

Reply:
Thanks for linking to that previous comment. I remember asking before but I never saw your reply for some reason. 

I don't like having the annotation because 
* It's noisy making me not be able to read the code faster. 
* Not needed and error prone because a typo in `@SerialName("this string here")` could introduce a bug. 
* Couples our code more to the kotlinx serialization library. If we were to swap to using a different library in the future, we would need to refactor a lot more of the code base.

Reply:
> It's noisy making me not be able to read the code faster

I think it is the other way round for me. Some parameters with annotations and some without means my eyeballs will keep dancing left right when reading a data class 😅 

```
data class Metric(
    @SerialName("delivery_id") val deliveryID: String,
    @SerialName("event") val event: MetricEvent,
    @SerialName("device_id") val deviceToken: String,
    @SerialName("timestamp") val timestamp: Long,
)
```

is much consistent and easier to read than the following in my opinion

```
data class Metric(
    @SerialName("delivery_id") val deliveryID: String,
    val event: MetricEvent,
    @SerialName("device_id") val deviceToken: String,
    val timestamp: Long,
)
```

> Not needed and error prone because a typo in @SerialName("this string here") could introduce a bug.

I think the probability of typo in both
```
@SerialName("plaftorm") val plaftorm: String
```
and 
```
val plaftorm: String
```
is practically the same. Because generally, most people would use the copy-paste technique here. Also, when fixing this, the former would be much easier than the latter in terms of refactoring as only fixing the annotation value can work.

> Couples our code more to the kotlinx serialization library. If we were to swap to using a different library in the future, we would need to refactor a lot more of the code base.

I think IDE makes it a lot more easier as we'll only need **Replace All** operation on `@SerialName`. 
And interestingly, imaging moving to other library and we find out that library requires us to add annotations explicitly to every parameter because it cannot guarantee the same correct name is being used when Kotlin class is converted to Objective-C or Swift. There is no limit of reasons for failure 😅

Reply:
I am open to ideas here. 

If having annotations on all properties is easier for you to read, then I am OK with having them. 

`Metric` is the only data class that we have where the property name does not equal the serial name. 

```kt
data class Metric(
    @SerialName("delivery_id") val deliveryID: String,
    val event: MetricEvent,
    @SerialName("device_id") val deviceToken: String,
    val timestamp: Long,
)
```

All other data classes have the property name match the JSON key making the data class not need any `@SerialName` annotations. That's the biggest reason why I would rather default to not using `@SerialName` on all properties. 1 data class needs them, all others do not. That sounds like a lot of boilerplate code to me. 

---

I wish that Kotlin allowed you to [create typealiases for annotations](https://discuss.kotlinlang.org/t/annotation-typealias/17784/3). Then, we could de-couple our data classes from the JSON library. 

We could create a typealias:

```kt
typealias JsonObject = Moshi.JsonClass(generateAdapter = true)
typealias JsonKey(val keyName: String) = Moshi.SerialName(name = keyName)
```

use those annotations in our data classes:
```kt
@JsonObject 
data class Metric(
  @JsonKey("foo")
  val foo: String 
}
```

Then, if we ever change libraries, we just need to modify the typealiases.

Reply:
Yeah, `typealias` can be really helpful for some cases. But unfortunately this doesn't look in their priority yet 😞

I'm open to ideas too, so once we pick this up later from list, we can get more opinions on this and go with whatever we decide.